File: app/src/main/java/com/byoutline/kickmaterial/utils/EnumDtoHelper.java
package com.byoutline.kickmaterial.utils;

import com.byoutline.kickmaterial.KickMaterialApp;

/**
 * @author Sebastian Kacprzak <nait at naitbit.com> on 11.03.2014
 */
public final class EnumDtoHelper {
    private EnumDtoHelper() {
    }

    public static String getDisplayName(int nameResId, String fallbackName) {
        KickMaterialApp instance = KickMaterialApp.component.getApp();
        if (instance == null) {
            return fallbackName;
        } else {
            return instance.getApplicationContext().getString(nameResId);
        }
    }
}

File: app/src/main/java/com/byoutline/kickmaterial/model/utils/EnumDtoHelper.kt
package com.byoutline.kickmaterial.model.utils

import com.byoutline.kickmaterial.KickMaterialApp

/**
 * @author Sebastian Kacprzak <nait at naitbit.com> on 11.03.2014
</nait> */
object EnumDtoHelper {

    fun getDisplayName(nameResId: Int, fallbackName: String): String {
        val instance = KickMaterialApp.component.app
        if (instance == null) {
            return fallbackName
        } else {
            return instance.applicationContext.getString(nameResId)
        }
    }
}
